### DISCUSSION TITLE
11.0.0 Preview 4

### DISCUSSION THREAD

**Discussion by danwalmsley** [AvaloniaUI/Avalonia#9417] [General]
<hr /><em>This discussion was created from the release <a href='https://github.com/AvaloniaUI/Avalonia/releases/tag/11.0.0-preview4'>11.0.0 Preview 4</a>.</em>

**Comment by olitee** [comment]
For anyone else having problems upgrading from Preview3 to Preview4, it's worth nothing that 

- **RichTextBlock** was renamed in Preview4 to **SelectableTextBlock**, 
- and Inlines support was also added to **TextBlock**.

Relevant commit: https://github.com/AvaloniaUI/Avalonia/commit/93b77b1b6a2181822f3bd1b074a504366aa4e0b4

**Reply by Gillibald** [reply]
Thanks

**Comment by Sigrec** [comment]
Upgraded from 0.10.18 and it broke CornerRadius for Border Control is there a change?

**Reply by timunie** [reply]
I think clipping of rounded corners are known to be broken, see #8916 If you faced any other issue, show us the code and at best a screen shot. Thx.

**Reply by Sigrec** [reply]
Going to downgrade back to stable release since I plan to release my application soon, I only upgraded because I really need the ProgressBar bug fix from this Ticket https://github.com/AvaloniaUI/Avalonia/pull/8488 and ColorPicker.

I installed the package version mentioned (0.10.999-cibuild0021834-beta) but it still didn't fix the issue, is there something I'm missing or did wrong thanks?

**Reply by timunie** [reply]
Yeah I think the fix wasn't backported and is only available as a preview. Sorry for that. If you can't upgrade, I'd just cooy the code of this control into your own project and apply the fix on your own.

**Reply by Sigrec** [reply]
I'll just upgrade then and find workarounds for other stuff, Thanks.

**Reply by Sigrec** [reply]
I forgot to mention that it seems like LineHeight property for TextBlock controls breaks in preview, is there a ticket already? I was trying to see if it was because I didn't set Height/Width for parent controls but doesn't seem to fix it.

**Reply by Gillibald** [reply]
No there is no issue for that. What is happening if you set the LineHeight?

**Reply by Sigrec** [reply]
> No there is no issue for that. What is happening if you set the LineHeight?

It doesn't apply the property at all for me, if I revert back to stable release, it applies it just fine.

**Comment by Splitwirez** [comment]
Just wanted to say, I'm really liking how 11.0.0 is shaping up thus far. `ControlTheme`s are...taking some getting used to, but are already proving *very* useful, definitely a highlight imo.

Things also seem a tad...snappier, in some areas? Are performance gains one of the expected perks of the new rendering stuff? Regardless of reason, I consider that much an absolute win.

Still definitely some jank here and there, so I'd imagine the 11.0.0 stable release is a bit of a ways off yet...but that future looks pretty bright so far :+1:

**Reply by grokys** [reply]
> Are performance gains one of the expected perks of the new rendering stuff?

Performance has been one of our main focuses for 11.0 - the renderer helps a lot but we've also done a lot of other performance improvements. For example if you open the ControlCatalog in the main repository and then go to the "Calendar" page, you can see a _huge_ difference between 0.10.0 and master.

**Comment by KindOfGod** [comment]
I upgraded from 0.10.18 and it broke the rounded Corners of the Apllication on Windows. It only happens when extending the ClientArea to the TitleBar. Is there a way to fix that without losing the design?

**Reply by robloo** [reply]
CornerRadius has some known issues at the moment; however, I think you found a new one. Personally I'm not aware of any fix or work-around for this one but you might want to create a new issue to discuss and track this.

**Reply by maxkatz6** [reply]
I am not sure if Windows.CornerRadious was ever applied on the window frame, as it was never implemented. And having to extend client area sounds as an expected behavior here. 
If you can prepare a minimal example and create an issue, it can be helpful.

**Comment by Sigrec** [comment]
Quick question related to the Avalonia ColorPicker, is it possible to remove the grey button on the right? Mainly I just want the button to open the colorpicker dialog to just be the color similar to the image.
![Screenshot 2023-01-21 085405](https://user-images.githubusercontent.com/53946709/213877854-3eb9e464-160c-410a-8422-21b35c635ea2.png)

**Reply by robloo** [reply]
Can you be more specific? I dont quite understand.

**Reply by Sigrec** [reply]
This is what it looks like by default, I want to remove the grey button on the right so it's just the color that I click to open the colorpicker.
![image](https://user-images.githubusercontent.com/53946709/213924858-5901b22e-fcfb-4cec-8ae6-2a6ce84573d6.png)

**Reply by robloo** [reply]
It's designed to be a drop down button (improved usability too) so no, getting rid of the drop down chevron and clickable button isn't supported as a property.

However, you can definitely do what you want if you

 1. Re-template the ColorPickerButton and remove the DropDownButton and use a Button instead.
 2. You can also just put a ColorView in a Button's flyout directly. This doesn't require making any new control but you won't be able to set properties quite as easily. You'll need to set properties on the ColorView directly.

**Reply by Sigrec** [reply]
I want to keep all the properties of the button both dropdown & clickable just have the grey button part not show, but I'll take a look at the re-template, thanks.

**Reply by robloo** [reply]
If you want to add it as a re-usable control in your app re-templating ColorPicker is the way to go. If you only want it in one view of your app I would just use ColorView in a button flyout.
